From aac8b8fccde3cc8a4fa5d55c747676d45a26df1b Mon Sep 17 00:00:00 2001
From: Mayank Sirotiya <Mayank.Sirotiya@imgtec.com>
Date: Mon, 27 Jul 2015 18:02:49 +0530
Subject: enable audio in marduk

Remove i2s pins which are used as gpio for providing
other functionalities on the Marduk board.

Change-Id: I9844f31cac6aa0a84c4c5a67ffbd4d1939ab22a3
Signed-off-by: Mayank Sirotiya <Mayank.Sirotiya@imgtec.com>
---
 arch/mips/boot/dts/pistachio/pistachio_marduk.dts | 77 +++++++++++++++++++++++
 1 file changed, 77 insertions(+)

diff --git a/arch/mips/boot/dts/pistachio/pistachio_marduk.dts b/arch/mips/boot/dts/pistachio/pistachio_marduk.dts
index a501b26..233d0cd 100644
--- a/arch/mips/boot/dts/pistachio/pistachio_marduk.dts
+++ b/arch/mips/boot/dts/pistachio/pistachio_marduk.dts
@@ -10,6 +10,7 @@
 /dts-v1/;
 
 #include "pistachio.dtsi"
+#include <dt-bindings/sound/pistachio-audio.h>
 
 / {
 	model = "IMG Marduk";
@@ -36,6 +37,43 @@
 		regulator-max-microvolt = <1800000>;
 		regulator-boot-on;
 	};
+
+	pistachio_audio_card {
+			compatible = "img,pistachio-audio";
+
+			clocks = <&clk_core CLK_AUDIO_PLL>,
+				 <&clk_core CLK_I2S>,
+				 <&clk_core CLK_AUDIO>;
+			clock-names = "audio_pll", "i2s_mclk", "dac_clk";
+
+			img,cr-periph = <&cr_periph>;
+
+			pinctrl-names = "default";
+			pinctrl-0 = <&dac_clk_pin>, <&i2s_mclk_pin>;
+
+			spdif-out {
+				cpu-dai = <&spdif_out>;
+			};
+
+			spdif-in {
+				cpu-dai = <&spdif_in>;
+			};
+
+			parallel-out {
+				cpu-dai = <&parallel_out>;
+				sound-dai = <&internal_dac>;
+			};
+
+			i2s-out {
+				cpu-dai = <&i2s_out>;
+				format = "i2s";
+			};
+
+			i2s-in {
+				cpu-dai = <&i2s_in>;
+				format = "i2s";
+			};
+		};
 };
 
 &spfi0 {
@@ -162,3 +200,42 @@
        num_streams = [02];
 };
 
+&i2s_out {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s_out_pins>;
+};
+
+&i2s_in {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s_in_pins>;
+};
+
+&i2s_out_pins {
+	i2s-out {
+		pins = "mfio37", "mfio38", "mfio39",
+		       "mfio40", "mfio41";
+		function = "i2s_out";
+	};
+};
+
+&i2s_in_pins {
+	i2s-in {
+		pins = "mfio47", "mfio48", "mfio49",
+		       "mfio50", "mfio51";
+		function = "i2s_in";
+	};
+};
+
+&spdif_out {
+	status = "okay";
+};
+
+&spdif_in {
+	status = "okay";
+};
+
+&parallel_out {
+	status = "okay";
+};
-- 
1.9.1

